### ISSUE TITLE
Installer does not update when using RAM drive for TEMP files on Windows with "Could not list processes locking resources. Result code is 5""

### ISSUE THREAD

**Issue by ramonsmits** [docker/for-win#13914]
### Description

I was affected by:

- https://github.com/docker/for-win/issues/13799

A workaround is to temporarily not use a RAM drive for TEMP files but the installer should either just work with such a configuration OR mention that is cannot work because of it.

### Reproduce

1. Setup RAM drive using ImDisk (likely other RAM drives have the same issue)
2. Map %TEMP% and %TMP% to your RAM drive (i.e. `R:\tmp`)
3. Update using the installer

### docker version

```bash
NA
```

### docker info

```bash
NA
```

### Diagnostics ID

NA

**Comment by alainbryden** [comment]
I'm affected by this too.

I'm honestly surprised this is an issue that needs special attention, I would have expected the use of a ram disk to be completely transparent. Perhaps the fact that we're using a RAM drive is a red-herring, and it's just related to remapping your temp folder to another disk?

For what it's worth, simply copying the file downloaded to `M:\Temp\DockerDesktopUpdates` to my Desktop (on C:) and re-running it from there (without re-mapping my temp folder) fixed the issue. So it's less about having the temp folder remapped, and more a matter of running the updater not-from the C drive.

**Comment by CherryDT** [comment]
I'm quite sure it's due to the RAM Drive, in particular ImDisk. The reason is that ImDisk doesn't register a volume with the windows volume manager (you won't see it in `mountvol` either). Some Windows API functions which use the volume manager will then not work with this volume, most importantly `GetFinalPathNameByHandle` (it returns `ERROR_INVALID_FUNCTION` then). There are also of course some libraries which use this function internally, such as libuv in `uv_fs_realpath` (which does have this limitation documented, however). This confuses some software, apparently including Docker.

You can use alternatives such as [this one](https://github.com/tmcdos/ramdisk).

**Comment by alainbryden** [comment]
I agree, I've since encountered several other tools (e.g. npm) which struggle with the default temp folder being in in ImDisk' RAM drive.

It's a shame, even the tool above doesn't have feature parity (most notably, a surprising inability to start out needing no RAM, and to dynamically grow its memory footprint with the size if the temp folder - it seems it must reserve memory for the entire max drive size in advance).

In any event, that's a "us problem". Can't fault developers for wanting to use modern file system features that ImDisk doesn't support. I just wish ram drives were more popular and better developed - they spare my ssd from so many write cycles.

**Comment by tophf** [comment]
In a [similar report](https://crbug.com/442094023) the suggested alternative for ImDisk is [AIM Toolkit](https://sourceforge.net/projects/aim-toolkit/) which includes the features of ImDisk.

**Comment by CherryDT** [comment]
> In any event, that's a "us problem". Can't fault developers for wanting to use modern file system features that ImDisk doesn't support.

Actually, I disagree - I can't find any documentation that mandates that every storage class _has to_ support the mount manager, in fact the [docs of `GetFinalPathNameByHandle`](https://learn.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-getfinalpathnamebyhandlew) even point out that not all drives will be and what to do in that case:

> Some third-party drivers can create a drive letter or mount point without using the Mount Manager. If the Mount Manager was not used to create the drive, then VOLUME_NAME_DOS or VOLUME_NAME_GUID will not succeed; only VOLUME_NAME_NT will be available. To determine the drive letter for the volume device path, use the QueryDosDevice function on every drive letter until a matching device name is found.

So, in my opinion, developers using that function should consider this situation and apply the solution recommended in the documentation cited above instead of mysteriously failing and stumping end users (especially those less technical than us who would never realize it's because of the RAM disk). This is especially true for libraries that wrap these WinAPI functions because I'd expect an established library to handle all those edge cases already for me (otherwise why would I even use a library in the first place), especially when they are documented. `libuv` at least recognizes that and has a note about it in their documentation even though they did not implement the solution themselves, but then again that means that a software like Docker (especially since that's not a one-man-in-a-garage project) ought to pick up on that and implement the solution themselves, in my opinion.

I mean: If I use some function and the docs say "may not work in case of X", I need to consider whether case X can happen and if it does, what the consequences/behavior would be in that case, and in this case I'd see that a) yes, X can happen (because the docs say some third-party drivers' usage may result in it so it's definitely a thing) and b) when it happens, the result is a crash and the user cannot install or uninstall my software, that makes it a critical problem for me so I'd build some safeguards in to not rely on that function alone and have another avenue to make the (un)installation succeed if that function fails. My two cents.
